shared/js/background/events/3p-tracking-cookie-blocking.js

```diff
@@ -52,26 +52,26 @@ function dropTracking3pCookiesFromRequest (request) {
     const tab = tabManager.get({ tabId: request.tabId })
     let requestHeaders = request.requestHeaders
 
-    if (tab && tab.site.isFeatureEnabled('trackingCookies3p') && request.type !== 'main_frame') {
-        if (!trackerutils.isTracker(request.url)) {
+    if (tab && request.type !== 'main_frame') {
+        const requestIsTracker = trackerutils.isTracker(request.url)
+        if (requestIsTracker && !tab.site.isFeatureEnabled('trackingCookies3p')) {
+            return { requestHeaders }
+        }
+        if (!requestIsTracker && !tab.site.isFeatureEnabled('nonTracking3pCookies')) {
```

Comment:
This is almost a copy/paste of then code above (from line 16), except with `requestHeaders` instead of `responseHeaders`. Could we remove the repetition?